### ISSUE TITLE
ValueError: failed to validate CustomSelect(id='p1024', ...).value: expected a value of type str, got 1 of type int

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#5560]
I'm on Panel 1.2.3 trying to help randomBloke in https://discourse.holoviz.org/t/inside-a-class-how-to-watch-for-changes-in-a-global-variable/6090/5.

When I serve the below

```python
import panel as pn
import param

class Component(param.Parameterized):
    select1 = param.Selector({"A": 1, "B": 2, "C": 3}, label="Select a letter")
    select2 = param.Selector({"A": 1, "B": 2, "C": 3}, label="Select a letter")

component = Component()

component.param.select2.objects = {"A": 1, "B": 2, "C": 3}

@pn.depends(component.param.select1, component.param.select2)
def result(value1, value2):
    return f"{value1}, {value2}"

pn.Column(component.param.select1, component.param.select2, result).servable()
```

I see.

![image](https://github.com/holoviz/panel/assets/42288570/ace553ef-b44b-4da7-a39f-078784ea99a2)

```bash
ValueError: failed to validate CustomSelect(id='p1024', ...).value: expected a value of type str, got 1 of type int
```

I would expect to be able to set `component.param.select2.objects = {"A": 1, "B": 2, "C": 3}` without raising an exception.

**Comment by aktech** [comment]
I just tried this on `panel==1.2.3` and also on `main`, it works fine for me:

```python
$ pip freeze | grep panel
panel==1.2.3
```

```bash
panel serve foo.py --autoreload
```

https://github.com/holoviz/panel/assets/5647941/fc1074a7-78ba-4b33-8025-e101466e425b

**Comment by philippjfr** [comment]
Indeed, cannot reproduce an issue here.

**Comment by philippjfr** [comment]
Ah this likely has been fixed by changes in Param 2.0.
